### PR TITLE
Remove Higher Kinded Types

### PR DESCRIPTION
**Pull request by nomisRev**
This PR removes `Kind` from the Arrow codebase.

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
@nomisRev 😘 👏 🥳 💌

File: arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt [hunk 1]
```diff
@@ -1,41 +1,9 @@
...
-)inline
-fun <A, T> ConstOf<A, T>.fix(): Const<A, T> =
-  this as Const<A, T>
-
-@Deprecated(
-  message = KindDeprecation,
-  level = DeprecationLevel.WARNING
-) fun <A, T> ConstOf<A, T>.value(): A = this.fix().value()
-
-data class Const<A, out T>(private val value: A) : ConstOf<A, T> {
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt:38]: I think we shouldn't remove this part, should we? Just the `ConstOf<A, T>` extension

File: arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt [hunk 2]
```diff
@@ -7,7 +7,6 @@ package arrow
   AnnotationTarget.FUNCTION
 )
 @MustBeDocumented
-@Deprecated(KindDeprecation)
 annotation class extension
```
Comment (aballano) [inline arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt:10]: This and `higherkind` got undeprecated?

File: arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt (hunk 1 above)
Comment (nomisRev) [reply arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt:38]: Only the `ConstOf<A, T>` extensions has been removed? 🤔 Am I missunderstanding your comment?

**Review by aballano — APPROVED** [review]
just one nit, looks good!

File: arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt (hunk 1 above)
Comment (franciscodr) [reply arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Const.kt:38]: Nevermind, I didn't see you were adding 
```
data class Const<A, out T>(private val value: A) {
```

File: arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt (hunk 2 above)
Comment (nomisRev) [reply arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt]: Twice in a row 🤦 3 times and I owe you a beer!

**Review by franciscodr — APPROVED** [review]
Thanks, @nomisRev!

File: arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt (hunk 2 above)
Comment (aballano) [reply arrow-libs/core/arrow-annotations/src/main/java/arrow/extension.kt]: 😘

**Pull request merged**